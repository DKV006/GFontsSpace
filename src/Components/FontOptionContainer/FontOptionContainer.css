.FontOptionContainer {
  border: 1px #dbdbdb solid;
  background-color: #fff;
  position: fixed;
  right: 0;
  width: 250px;
  padding: 20px 20px 8px 20px; }
  .FontOptionContainer .FontOptionContainer__Item {
    padding-bottom: 12px; }
  .FontOptionContainer .FontOptionContainer__Title {
    font-family: 'Oswald', sans-serif;
    text-transform: uppercase;
    margin-bottom: 6px; }
  .FontOptionContainer ul {
    display: flex;
    flex-direction: column; }
    .FontOptionContainer ul li {
      border: 3px #ffdf01 solid;
      border-radius: 3px;
      padding: 4px 5px;
      display: flex;
      flex: 1;
      justify-content: center;
      margin-bottom: 10px;
      cursor: pointer;
      text-transform: capitalize; }
      .FontOptionContainer ul li:last-child {
        margin-right: 0; }
      .FontOptionContainer ul li:hover {
        background: #000000;
        color: #ffffff; }
      .FontOptionContainer ul li.active {
        background: #ffdf01;
        color: #000; }
  .FontOptionContainer .FontOptionContainer__Properties {
    display: flex;
    flex-direction: column; }
    .FontOptionContainer .FontOptionContainer__Properties .Select {
      width: 100%;
      margin-bottom: 10px; }
    .FontOptionContainer .FontOptionContainer__Properties .OtherProperties {
      display: flex; }
    .FontOptionContainer .FontOptionContainer__Properties .FontSize__Option input[type="number"] {
      width: 45px;
      height: 36px;
      text-align: center;
      -webkit-appearance: none;
      border: none;
      background: none;
      border: 1px #ccc solid;
      border-radius: 4px;
      margin-right: 10px;
      font-size: inherit; }
      .FontOptionContainer .FontOptionContainer__Properties .FontSize__Option input[type="number"]:focus {
        outline: none; }
    .FontOptionContainer .FontOptionContainer__Properties .ColorOption__BackGround {
      margin-right: 10px; }
      .FontOptionContainer .FontOptionContainer__Properties .ColorOption__BackGround .ColorPicker .ColorPicker__Color {
        box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.1); }
    .FontOptionContainer .FontOptionContainer__Properties .ColorOption__ForeGround {
      margin-right: 10px;
      font-size: 32px;
      text-align: center; }
