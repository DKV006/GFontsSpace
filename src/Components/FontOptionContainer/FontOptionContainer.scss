.FontOptionContainer {
  border: 1px #dbdbdb solid;
  position: fixed;
  width: 100%;
  max-width: 1000px;
  padding: 20px;

  .FontOptionContainer__Item {
    padding-bottom: 12px;
  }

  .FontOptionContainer__Title {
    font-family: 'Oswald', sans-serif;
    text-transform: uppercase;
    margin-bottom: 6px;
  }
  

  ul {
    display: flex;
    li {
      border: 3px #ffdf01 solid;
      border-radius: 3px;
      padding: 4px 5px;
      display: flex;
      flex: 1;
      justify-content: center;
      margin-right: 10px;
      cursor: pointer;

      &:last-child {
        margin-right: 0;
      }

      &:hover {
        background: #000000;
        color: #ffffff
      }

      &.active {
        background: #ffdf01;
      }
    }
  }

}