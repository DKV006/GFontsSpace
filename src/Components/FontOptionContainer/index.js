import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';
import 'react-select/dist/react-select.css'

import reactCSS from 'reactcss';
import { SketchPicker } from 'react-color';
import WebFont from 'webfontloader';
import './FontOptionContainer.css'

class FontOptionContainer extends Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedFontFamily: '',
            selectedFontVariant: '',
            displayColorPicker: false,
            clearable: true,
        }

        this.handleChangeFontFamily = this.handleChangeFontFamily.bind(this);
        this.handleChangeFontVariant = this.handleChangeFontVariant.bind(this);
    }

    handleChangeFontFamily = (selectedFontFamily) => {
        this.setState({ selectedFontFamily },()=>{
            selectedFontFamily && WebFont.load({
                google: {
                    families: [selectedFontFamily.value]
                }
            });
        });
        this.props.handleChangeFontFamily(selectedFontFamily)
    }

    handleChangeFontVariant = (selectedFontVariant) => {
        this.setState({ selectedFontVariant });
        this.props.handleChangeFontVariant(selectedFontVariant)
    }

    render() {
        return (
            <section className="FontOptionContainer">
                <section className="FontOptionContainer__Item">
                    <h4 className="FontOptionContainer__Title">Classification</h4>
                    <ul>
                        {this.props.categories.map((category, index) =>
                            <li key={`${category.name}-${index}`}
                                data-category={category.name}
                                className={`${category.isActive ? "active" : "inactive"}`}
                                onClick={category.handleClickOnCategory}>
                                {category.name}
                            </li>)
                        }
                    </ul>
                </section>
                <section className="FontOptionContainer__Item">
                    <h4 className="FontOptionContainer__Title">Propeties</h4>
                    <div className="FontOptionContainer__Properties">
                        <Select
                            name="FontFamilySelectBox"
                            value={this.state.selectedFontFamily}
                            onChange={this.handleChangeFontFamily}
                            options={this.props.fontFamilies}
                            clearable={this.state.clearable}
                            placeholder="Select Font Family"
                        />

                        <Select
                            name="FontVariantSelectBox"
                            value={this.state.selectedFontVariant}
                            onChange={this.handleChangeFontVariant}
                            options={this.props.fontVariants}
                            clearable={this.state.clearable}
                            placeholder="Select Font Variants"
                        />
                        <div className="OtherProperties">
                            <div className="FontSize__Option">
                                <input type="number" name="quantity" min="6" max="248" value="16" />
                            </div>
                            <div className="ColorOption__ForeGround">
                                A
                            </div>

                            <div className="ColorOption__BackGround"></div>
                        </div>
                        
                    </div>
                </section>
            </section>
        );
    }
}

FontOptionContainer.propTypes = {

};

export default FontOptionContainer;