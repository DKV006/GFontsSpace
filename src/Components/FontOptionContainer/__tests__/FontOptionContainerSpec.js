import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import FontOptionContainer from './../index';

describe('FontOptionContainer', () => {
    it('renders correctly', () => {
        const fontOptionContainerProp = {
            categories:[{
                name: "sarif",
                isActive: true,
                handleClickOnCategory: jest.fn()
            }],
            fontFamilies:[
                {
                    value :"some font",
                    label :"some font",
                }
            ],
            handleChangeFontFamily:jest.fn(),
            fontVariants:["regular", 400],
            handleChangeFontVariant:jest.fn(),
            color:{
                r: '0',
                g: '0',
                b: '0',
                a: '1',
              },
            handleChangeColor:jest.fn(),
            bgColor:{
                r: '255',
                g: '255',
                b: '255',
                a: '1',
              },
        }
        let tree = renderer.create(
            <FontOptionContainer {...fontOptionContainerProp}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
