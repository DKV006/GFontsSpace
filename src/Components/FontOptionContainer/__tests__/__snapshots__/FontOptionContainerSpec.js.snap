// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontOptionContainer renders correctly 1`] = `
<section
  className="FontOptionContainer"
>
  <section
    className="FontOptionContainer__Item"
  >
    <h4
      className="FontOptionContainer__Title"
    >
      Classification
    </h4>
    <ul>
      <li
        className="active"
        data-category="sarif"
        onClick={[Function]}
      >
        sarif
      </li>
    </ul>
  </section>
  <section
    className="FontOptionContainer__Item"
  >
    <h4
      className="FontOptionContainer__Title"
    >
      Propeties
    </h4>
    <div
      className="FontOptionContainer__Properties"
    >
      <div
        className="Select Select--single is-clearable is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              className="Select-placeholder"
            >
              Select Font Family
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-activedescendant="react-select-2--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <div
        className="Select Select--single is-clearable is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-3--value"
          >
            <div
              className="Select-placeholder"
            >
              Select Font Variants
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-activedescendant="react-select-3--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <div
        className="OtherProperties"
      >
        <div
          className="FontSize__Option"
        >
          <input
            max="248"
            min="6"
            name="quantity"
            type="number"
            value="16"
          />
        </div>
        <div
          className="ColorOption__ForeGround"
        >
          <div
            className="ColorPicker"
          >
            <div
              className="ColorPicker__Swatch"
              onClick={[Function]}
            >
              <div
                className="ColorPicker__Color"
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 1)",
                  }
                }
              >
                A
              </div>
            </div>
          </div>
        </div>
        <div
          className="ColorOption__BackGround"
        >
          <div
            className="ColorPicker"
          >
            <div
              className="ColorPicker__Swatch"
              onClick={[Function]}
            >
              <div
                className="ColorPicker__Color"
                style={
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 1)",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;
